src/xtgeo/surface/regular_surface.py

```diff
@@ -235,135 +224,52 @@ def __init__(
-            )
+        self._values = None
+        if values is None:
+            values = np.ma.zeros((self._ncol, self._nrow))
+        self.values = values
 
-        if self._ilines is None:
+        if ilines is None:
             self._ilines = np.array(range(1, self._ncol + 1), dtype=np.int32)
             self._xlines = np.array(range(1, self._nrow + 1), dtype=np.int32)
+        else:
+            self._ilines = ilines
+            self._xlines = xlines
 
-        if self._values is not None and self._values.mask is ma.nomask:
-            self._values = ma.array(self._values, mask=ma.getmaskarray(self._values))
-
-        self._name = name
         self._masked = masked  # TODO: check usecase
         self._metadata.required = self
         logger.info("Ran __init__ method for RegularSurface object %s", id(self))
 
-    def _instance_from_filelike(self, sfile, template, fformat, engine, values):
-        """Detect input automatically if possible and load to make the instance."""
-        sobj = xtgeo._XTGeoFile(sfile)
-
-        loadvalues = True
-        if isinstance(values, bool) and values is False:
-            loadvalues = False
-
-        if not sobj.memstream:
-            if sobj.file.suffix == "hdf":
-                self.from_hdf(sfile, values=loadvalues)
-            else:
-                self.from_file(
-                    sfile,
-                    fformat=fformat,
-                    engine=engine,
-                    template=template,
-                    values=loadvalues,
-                )
-        else:
-            if fformat == "hdf":
-                self.from_hdf(sfile, values=loadvalues)
-            else:
-                self.from_file(sfile, fformat=fformat, values=loadvalues)
-
-    def _instance_from_spec(
-        self, ncol, nrow, xori, yori, xinc, yinc, yflip, rotation, values, **kwargs
-    ):
-        # make instance by kw spesification
-        self._xori = xori
-        self._yori = yori
-
-        # backward compatibility
-        if "nx" in kwargs:
-            self._ncol = kwargs.get("nx", 5)
-            self._nrow = kwargs.get("ny", 3)
-        else:
-            self._ncol = ncol
-            self._nrow = nrow
-
-        self._xinc = xinc
-        self._yinc = yinc
-        self._rotation = rotation
-
-        default = ncol == 5 and nrow == 3 and xori == yori == 0.0 and xinc == yinc == 25
-
-        if values is None and default:
-
-            # make default surface (mostly for unit testing)
-            values = np.array(
-                [[1, 6, 11], [2, 7, 12], [3, 8, 1e33], [4, 9, 14], [5, 10, 15]],
-                dtype=np.float64,
-                order="C",
-            )
-            # make it masked
-            values = ma.masked_greater(values, self.undef_limit)
-            self._masked = True
-            self._values = values
-
-        elif values is None and not default:
-            self._values = np.ma.zeros((self._ncol, self._nrow))
```

Comment:
Why is the behavior different between initializing from spec and file here? If I initialize from a file without values, I get `self.values == None`, if I initialize from spec without values I get `np.values == np.ma.zeros(...)`?